JsonPatch.Tests/ExtensionTests.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Json.Patch;
+using Json.Schema;
+using NUnit.Framework;
+using NUnit.Framework.Internal.Execution;
+
+namespace JsonPatch.Tests
+{
+	public class ExtensionTests
+	{
+		[Test]
+		public void CreatePatch_Test()
+		{
+			var e1 = new TestModel()
+			{
+				Id = Guid.NewGuid(),
+				Attributes = JsonDocument.Parse("[{\"test\":\"test123\"},{\"test\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,4]}]")
+			};
+			var e2 = new TestModel()
+			{
+				Id = Guid.NewGuid(),
+				Attributes = JsonDocument.Parse("[{\"test\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,3]}]")
+			};
+			var op = e1.CreatePatch(e2);
+			Assert.AreEqual(op.Operations.Count, 6);
+
+			Assert.AreNotEqual(e1.Id, e2.Id);
+			Assert.AreNotEqual(e1.Attributes.RootElement.ToString(), e2.Attributes.RootElement.ToString());
+
+			e1 = op.ApplyPatch(e1);
+
+			Assert.AreEqual(e1.Id, e2.Id);
+			Assert.AreEqual(e1.Attributes.RootElement.ToString(), e2.Attributes.RootElement.ToString());
+		}
+
+		[Test]
+		public void Add_Test()
```

Comment:
These tests aren't quite what I'm looking for.  They should generate the patch and compare that to an expected patch.

```c#
public void Add()
{
    var initial = // initial model;
    var target = // target model;
    var expectedPatch = // explicitly create the expected JsonPatch object

    var actualPatch = initial.CreatePatch(target);

    Assert.AreEqual(expectedPatch, actualPatch);
}
```

This is a more typical unit test pattern.

Reply:
Well in most cases I'll agree, but what with complex objects? It's a bit hard to create large patch manually. By the logic if patch was correct so objects would be equal, isn't it?

Reply:
This comment is for all of the simple tests.  I want to ensure that the patch operations are generated correctly.  To do that you need to analyze the patch itself.

The more complex tests (e.g. the round trip) can stay as it is, but these simple ones need to change.